src/resolver/apidom/reference/dereference/strategies/openapi-3-1-swagger-client/visitors/all-of.js

```diff
@@ -37,7 +43,32 @@ class AllOfVisitor {
       while (schemaElement.hasKey('allOf')) {
         const { allOf } = schemaElement;
         schemaElement.remove('allOf');
-        const allOfMerged = deepmerge.all([...allOf.content, schemaElement]);
+        const allOfMerged = deepmerge.all([...allOf.content, schemaElement], {
+          customMerge: (keyElement) => {
+            if (toValue(keyElement) === 'enum') {
+              return (targetElement, sourceElement) => {
+                if (isArrayElement(targetElement) && isArrayElement(sourceElement)) {
+                  const primitiveElements = new ArrayElement([
+                    ...targetElement.findElements(isPrimitiveElement),
```

Comment:
`isPrimitiveElement` is more about distinction between generic APiDOM elements and Semantic ones. `info`  (from OpenAPI spec) vs `object`.  So this doesn't really do what is was intended to do. 

Let's try to put all element inside one array, filter out nonsense (object/array), and use R.uniqBy with `toValue` as first argument.